File: soske/data_test.go
package soske

import (
	"os"
	"testing"

	_ "github.com/mattn/go-sqlite3"
	"github.com/stretchr/testify/assert"
)

func TestDbOpen(t *testing.T) {
	// setup
	file, _ := os.CreateTemp("", "soske-test-*.db")
	defer file.Close()
	defer os.Remove(file.Name())

	// success - pragma & schema
	var num int
	db := DbOpen(file.Name(), Pragma, "create table Test(foo)")
	db.Get(&num, "select count(*) from SQLITE_SCHEMA")
	assert.Equal(t, 1, num)
	assert.NotNil(t, db)

	// success - pragma only
	db = DbOpen(file.Name(), Pragma, "drop table Test")
	db.Get(&num, "select count(*) from SQLITE_SCHEMA")
	assert.Equal(t, 1, num)
	assert.NotNil(t, db)
}

func TestDbExecute(t *testing.T) {
	// setup
	db := TestDB()

	// success
	var str string
	DbExecute(db, "insert into Keys (name) values (?)", "test")
	db.Get(&str, "select name from Keys order by init desc", "test")
	assert.Equal(t, "test", str)
}

func TestDbInteger(t *testing.T) {
	// setup
	db := TestDB()

	// success
	num := DbInteger(db, "select init from Keys where name=?", "alpha")
	assert.Equal(t, 1000, num)
}

func TestDbString(t *testing.T) {
	// setuo
	db := TestDB()

	// success
	str := DbString(db, "select name from Keys where name=?", "alpha")
	assert.Equal(t, "alpha", str)
}

func TestDbStrings(t *testing.T) {
	// setuo
	db := TestDB()

	// success
	strs := DbStrings(db, "select name from Keys where name=?", "alpha")
	assert.Equal(t, []string{"alpha"}, strs)
}

File: soske/tools/data_test.py
"""
Tests for 'soske.tools.data'.
"""

from soske.tools import data, sqls
from soske.tools.test import conn


def test_open(tmp_path):
    # setup
    path = str(tmp_path / "test.db")

    # success - pragma & schema
    conn = data.open(path, sqls.PRAGMA, "create table Tests(t)")
    curs = conn.execute("select count(*) from SQLITE_SCHEMA")
    assert curs.fetchone()["count(*)"] != 0

    # success - pragma only
    conn = data.open(path, sqls.PRAGMA, "drop table Tests")
    curs = conn.execute("select count(*) from SQLITE_SCHEMA")
    assert curs.fetchone()["count(*)"] != 0


def test_exec(conn):
    # success
    data.exec(conn, "insert into Keys values ('test', false, 123)")
    curs = conn.execute("select * from Keys where name=?", ["test"])
    assert curs.fetchone()["name"] == "test"


def test_row(conn):
    # success
    row = data.row(conn, "select * from Keys where name=?", "alpha")
    assert row["name"] == "alpha"


def test_rows(conn):
    # success
    rows = data.rows(conn, "select * from Keys where name=?", "alpha")
    assert [row["name"] for row in rows] == ["alpha"]
